### PR TITLE
Make VentMode_PRESSURE_CONTROL work.

### PR DESCRIPTION
**Pull request by jlebar**
<git-pr-chain>

#### Commits in this PR
1. Make VentMode_PRESSURE_CONTROL work.
    
    Fix a bug and add a test.

#### [PR chain](https://github.com/jlebar/git-pr-chain)
1. #205 Clarify comment on sensor scaling factor.
1. 👉 #211 Make VentMode_PRESSURE_CONTROL work. 👈 **YOU ARE HERE**
1. #212 Fix pressure sensors pinout.
1. #216 Explicitly set the output limits in our blower PID.
1. #217 Control the solenoid from the Blower FSM.
1. #218 Add NO_GUI_DEV_MODE, for running the controller without a GUI.
1. #219 Tune PID.

</git-pr-chain>

### CODE REVIEW

File: controller/test/blower_fsm/blower_fsm_test.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+TEST_F(BlowerFsmTest, PressureControl) {
+  VentParams p = VentParams_init_zero;
+  p.mode = VentMode_PRESSURE_CONTROL;
+  // 20 breaths/min = 3s/breath.  I:E = 2 means 2s for inspire, 1s for expire.
+  p.breaths_per_min = 20;
+  p.inspiratory_expiratory_ratio = 2;
+  p.peep_cm_h2o = 10;
+  p.pip_cm_h2o = 20;
+
+  EXPECT_EQ(blower_fsm_get_setpoint(p).cmH2O(), 20);
```
Comment (jkff) [inline controller/test/blower_fsm/blower_fsm_test.cpp:47]: I'd structure this a little differently but fine as is too:

```
vector<tuple</*time_millis=*/int, /*setpoint=*/float>> expected =
    {{0, 20}, {1000, 20}, {1999, 20}, ...};
```
and iterate over it with sleeping for the delta between neighboring points.

File: controller/test/blower_fsm/blower_fsm_test.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  Hal.delay(2); // t = 2001
+  EXPECT_EQ(blower_fsm_get_setpoint(p).cmH2O(), 10);
+  Hal.delay(998); // t = 2999
+  EXPECT_EQ(blower_fsm_get_setpoint(p).cmH2O(), 10);
+
+  Hal.delay(2); // t = 3001
+  EXPECT_EQ(blower_fsm_get_setpoint(p).cmH2O(), 20);
+}
```
Comment (jkff) [inline controller/test/blower_fsm/blower_fsm_test.cpp:60]: Any TODOs you can think of in terms of what else needs to be tested, so others can pick up?

File: controller/test/blower_fsm/blower_fsm_test.cpp (hunk 1 above)
Comment (jlebar) [reply controller/test/blower_fsm/blower_fsm_test.cpp:47]: Great idea.  Factored it to make it easier for someone to contribute additional tests, too.

File: controller/test/blower_fsm/blower_fsm_test.cpp (hunk 2 above)
Comment (jlebar) [reply controller/test/blower_fsm/blower_fsm_test.cpp:60]: https://github.com/RespiraWorks/VentilatorSoftware/issues/220

File: controller/test/blower_fsm/blower_fsm_test.cpp (hunk 1 above)
Comment (jlebar) [reply controller/test/blower_fsm/blower_fsm_test.cpp:47]: Going to make this change in #217 because it's annoying to move down here.

**Comment by jlebar** [comment]
Somehow github closed this PR during a git-pr-chain operation.  Not totally sure what happened.  I'm going to force-merge https://github.com/RespiraWorks/VentilatorSoftware/pull/222, which is identical to this.

**Pull request closed without merging**